src/passes/RemoveNonJSOps.cpp

```diff
@@ -14,251 +14,137 @@
-    Index rem = 4;
-
-    UnaryOp ceilOp = CeilFloat32;
-    UnaryOp floorOp = FloorFloat32;
-    BinaryOp subOp = SubFloat32;
-    BinaryOp ltOp = LtFloat32;
-    BinaryOp gtOp = GtFloat32;
-    BinaryOp divOp = DivFloat32;
-    BinaryOp eqOp = EqFloat32;
-    Literal litHalf((float) 0.5);
-    Literal litOne((float) 1.0);
-    Literal litZero((float) 0.0);
-    Literal litTwo((float) 2.0);
-    if (f == f64) {
-      ceilOp = CeilFloat64;
-      floorOp = FloorFloat64;
-      subOp = SubFloat64;
-      ltOp = LtFloat64;
-      gtOp = GtFloat64;
-      divOp = DivFloat64;
-      eqOp = EqFloat64;
-      litHalf = Literal((double) 0.5);
-      litOne = Literal((double) 1.0);
-      litZero = Literal((double) 0.0);
-      litTwo = Literal((double) 2.0);
+  void addNeededFunctions(Module &m, Name name, std::set<Name> &needed) {
+    if (needed.count(name)) {
+      return;
     }
+    needed.insert(name);
 
-    Expression *body = builder->blockify(
-      builder->makeSetLocal(
-        ceil,
-        builder->makeUnary(ceilOp, builder->makeGetLocal(arg, f))
-      ),
-      builder->makeSetLocal(
-        floor,
-        builder->makeUnary(floorOp, builder->makeGetLocal(arg, f))
-      ),
-      builder->makeSetLocal(
-        fract,
-        builder->makeBinary(
-          subOp,
-          builder->makeGetLocal(arg, f),
-          builder->makeGetLocal(floor, f)
-        )
-      ),
-      builder->makeIf(
-        builder->makeBinary(
-          ltOp,
-          builder->makeGetLocal(fract, f),
-          builder->makeConst(litHalf)
-        ),
-        builder->makeGetLocal(floor, f),
-        builder->makeIf(
-          builder->makeBinary(
-            gtOp,
-            builder->makeGetLocal(fract, f),
-            builder->makeConst(litHalf)
-          ),
-          builder->makeGetLocal(ceil, f),
-          builder->blockify(
-            builder->makeSetLocal(
-              rem,
-              builder->makeBinary(
-                divOp,
-                builder->makeGetLocal(floor, f),
-                builder->makeConst(litTwo)
-              )
-            ),
-            builder->makeIf(
-              builder->makeBinary(
-                eqOp,
-                builder->makeBinary(
-                  subOp,
-                  builder->makeGetLocal(rem, f),
-                  builder->makeUnary(
-                    floorOp,
-                    builder->makeGetLocal(rem, f)
-                  )
-                ),
-                builder->makeConst(litZero)
-              ),
-              builder->makeGetLocal(floor, f),
-              builder->makeGetLocal(ceil, f)
-            )
-          )
-        )
-      )
-    );
-    std::vector<Type> params = {f};
-    std::vector<Type> vars = {f, f, f, f, f};
-    Name name = f == f32 ? WASM_NEAREST_F32 : WASM_NEAREST_F64;
-    return builder->makeFunction(name, std::move(params), f, std::move(vars), body);
-  }
+    struct CallFinder: public PostWalker<CallFinder> {
```

Comment:
instead of CallFinder, see `src/ir/find_all.h` which you can use to get all the Calls.